### PR TITLE
[Fixes #5158] Fix thesaurus autocomplete

### LINKED ISSUE #5158: Fix Thesaurus autocomplete
## Expected Behavior
It should be possible to set Thesauri in settings and import Thesaurus-keywords  as described here:
https://github.com/geosolutions-it/geonode/wiki/Thesaurus-keywords

## Actual Behavior
The mentioned link https://raw.githubusercontent.com/geonetwork/util-gemet/master/thesauri/inspire-theme.rdf does not exist anymore. From what I read it has been moved to http://inspire.ec.europa.eu/theme (https://github.com/geonetwork/core-geonetwork/issues/2506)

Using a scheme from inspire.ec.europa.eu to import to geonode fails. It looks the structure has changed and the management command load_thesaurus  needs to be updated to deal with the new scheme.

## Steps to Reproduce the Problem

  1. enable thesauri in settings
  2. try to follow the instructions from above link

## Specifications

  - GeoNode version: master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker): manual
  - Platform: osx

### PR DESCRIPTION
**Pull request by afabiani**
- Fixes #5158 : Update thesaurus management command and fix autocomplete

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there are explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by afabiani** [comment]
@t-book still missing documentation and (possibly) some tests

**Comment by t-book** [comment]
@afabiani I've docs ready. can contribute tomorrow. thanks!

**Comment by t-book** [comment]
😳
Bildschirmfoto 2019-10-31 um 22 51 21

**Comment by t-book** [comment]
Looks good @afabiani I've just recognized when setting some thesauri on a layer resource and you revisit the wizard again it fails with:

Bildschirmfoto 2019-11-01 um 09 00 20

File: geonode/layers/views.py [hunk 1]
```diff
@@ -991,35 +990,35 @@ def layer_metadata(
...
+                el = settings.THESAURUS
+                thesaurus_name = el['name']
+                try:
+                    t = Thesaurus.objects.get(identifier=thesaurus_name)
+                    for tk in t.thesaurus.filter(pk__in=tkeywords_ids):
+                        tkl = tk.keyword.filter(lang=lang)
+                        if len(tkl) > 0:
+                            tkl_ids = ",".join(
+                                map(str, tkl.values_list('id', flat=True)))
+                            tkeywords_list += "," + \
```
Comment (t-book) [inline geonode/layers/views.py]: @afabiani regarding the described error from the PR comments, it think this should be a list? 

`tkeywords_list = [tkl_ids if len(tkeywords_list) > 0 else tkl_ids]
`

Comment (t-book) [reply geonode/layers/views.py]: while it solved the error. It will not update/remove the ids anymore ... mhhh

**Comment by afabiani** [comment]
@t-book fixed, thanks. Waiting for Travis to be happy!

**Comment by afabiani** [comment]
> Dear @afabiani, thanks a lot! This did it! The only thing I would change is that _resourcebase_info_panel is showing the tkeywords only if THESAURUS is enabled and adding commas between words.
> 
> **From:**
> https://github.com/GeoNode/geonode/blob/d2fa2b0669e67009604544b609692f4895d55405/geonode/base/templates/base/_resourcebase_info_panel.html#L58-L69
> 
> **To:**
> 
> ```
>       {% if THESAURI_FILTERS %}
>         {% if resource.tkeywords.count > 0 %}
>           {% get_current_language as LANGUAGE_CODE %}
>           {% for keyword in resource.tkeywords.all %}
>             <a itemscope itemtype="http://schema.org/Text" href="{% url "search" %}?tkeywords__id__in={{ keyword.id }}">
>               {% for label in keyword.labels %}
>                 {% if forloop.first %},{% endif %}
>                 {% if label.lang == LANGUAGE_CODE %}
>                   {{ label }}
>                 {% endif %}
>               {% endfor %}
>             </a>
>           {% endfor %}
>         {% endif %}
>       {% endif %}
> ```
> 
> What do you think?

@t-book sure, you are right. Thanks for the hint. I'll push the change to the PR accordingly.

**Pull request merged**